pymodbus/client/asynchronous/factory/tcp.py

```diff
@@ -105,6 +105,25 @@ def async_io_factory(host="127.0.0.1", port=Defaults.Port, framer=None,
     return loop, client
 
 
+def trio_factory(host="127.0.0.1", port=Defaults.Port, framer=None,
+                 source_address=None, timeout=None, **kwargs):
+    """
+    Factory to create Trio based asynchronous tcp clients
+    :param host: Host IP address
+    :param port: Port
+    :param framer: Modbus Framer
+    :param source_address: Bind address
+    :param timeout: Timeout in seconds
+    :param kwargs:
+    :return: tcp client
+    """
+    from pymodbus.client.asynchronous.trio import init_tcp_client
+    proto_cls = kwargs.get("proto_cls", None)
+    client = init_tcp_client(proto_cls, host, port)
+
+    return client
```

Comment:
This is inconsistent with other factories which return two values.

Reply:
They do return two values, but I don't see the consistency in the meaning of those two values.  Can you help me understand what sort of thing the first value should be and what the second should be?  Then I can try to map that into the Trio stuff.

Twisted: protocol / deferred
Tornado: protocol / future
asyncio: event loop / client
Trio: ? / ?

I can see the similarity between Twisted and Tornado.  I don't see how asyncio fits.  Apparently when I coded this a couple months ago I didn't see what other useful thing I could return in the Trio case.  We'll see if I can find something now.

Reply:
Relatedly, the following docstring seems to only get asyncio right.

https://github.com/riptideio/pymodbus/blob/c7a582b6796be211d693dba80367f2b26846c134/pymodbus/client/asynchronous/__init__.py#L15-L22